src/map/handler/Map.BoxZoom.js

```diff
@@ -26,6 +26,17 @@ L.Map.BoxZoom = L.Handler.extend({
 		return this._moved;
 	},
 
+	abort: function () {
+		this._resetState();
+		this._finish();
+	},
+
+	_resetState: function () {
+		// Postpone to next JS tick so internal click event handling
+		// still see it as "moved".
+		setTimeout(L.bind(this._resetState, this), 0);
```

Comment:
this looks weird, isn't it going to an infinite loop? where is `this._moved = false` part?